### ISSUE TITLE
Admin API should include WWW-Authenticate header in 401 responses

### ISSUE THREAD

**Issue by pschichtel** [keycloak/keycloak#9386]
### Description

Currently the admin API does not include the WWW-Authenticate header in 401 responses as per [RFC 2617](https://datatracker.ietf.org/doc/html/rfc2617#section-1.2) (superceded by [RFC 7235](https://datatracker.ietf.org/doc/html/rfc7235#section-3.1)) (MUST for 401 responses) and [RFC 6750](https://datatracker.ietf.org/doc/html/rfc6750#section-3) (MUST). For improved client library support I suggest to add this.

### Motivation

I've been writing a small keycloak admin API client based on kotlin and the [ktor](https://ktor.io) HTTP client library, that uses service accounts with minimal privileges. The ktor includes functionality for automatic access token retrieval and refreshing. Since it does support multiple authentication mechanisms at once, it relies on the WWW-Authenticate header to select an appropriate mechanism. Without the WWW-Authenticate the token refresh does not work.

The lack of control about when the token is refreshed can be considered a ktor bug/feature-request, however I still think it is valid to rely on the WWW-Authenticate header, that is why I'm opening the request here.

### Details

Since the admin API currently supports both OAuth and basic auth, I guess a valid response would be something like (for the master realm):

```
WWW-Authenticate: Basic realm="master", Bearer realm="master"
```

The [oauth2 RFC's section 3](https://datatracker.ietf.org/doc/html/rfc6750#section-3) documents a second parameter besides realm that could be added to the Bearer header: `scope`. This would be useful for clients to automatically know the required scopes of an endpoint, however to satisfy this request and the RFC this is not strictly necessary.

**Comment by emilv-barium** [comment]
I agree that this would be correct behavior from Keycloak Admin API.

Your problem also seems to be fixed with this Ktor commit: https://github.com/ktorio/ktor/commit/a9e4f12bdf071c28ca8c7b7648103be8b6817081 (not released yet as of v 1.6.8, target release 2.0.0)

**Comment by pschichtel** [comment]
Yes that commit indeed fixes my concrete issue, thanks for the hint.
